packages/common/src/util.ts

```diff
@@ -137,6 +138,26 @@ export function countSignatures(
     return totalSignatureCount;
 }
 
+/**
+ * Converts a wasm module to a smart contract schema.
+ * @param wasm the wasm module as a Buffer
+ * @returns the smart contract schema as a Buffer
+ */
+export function wasmToSchema(wasm: Buffer): Buffer {
```

Comment:
I think it could make sense to test this and check the "correct" schema is extracted.
I.e. without the node, where we have an expected schema to check against.

otherwise LGTM.

Reply:
Good suggestion

Reply:
Done, take a look. Note that I found a bug in `getModuleSource` while working on this: #173